gammapy/analysis/steps.py

```diff
@@ -0,0 +1,498 @@
+
+        self.analysis.observations = self.analysis.datastore.get_observations(
+            ids, skip_missing=True, required_irf=observations_settings.required_irf
+        )
+
+        if observations_settings.obs_time.start is not None:
+            start = observations_settings.obs_time.start
+            stop = observations_settings.obs_time.stop
+            if len(start.shape) == 0:
+                time_intervals = [(start, stop)]
+            else:
+                time_intervals = [(tstart, tstop) for tstart, tstop in zip(start, stop)]
+            self.analysis.observations = self.analysis.observations.select_time(time_intervals)
+
+        for obs in self.analysis.observations:
+            self.analysis.log.debug(obs)
+
+    def _set_data_store(self):
+        """Set the datastore on the Analysis object."""
+        path = make_path(self.analysis.config.observations.datastore)
+        if path.is_file():
+            self.analysis.log.debug(f"Setting datastore from file: {path}")
+            self.analysis.datastore = DataStore.from_file(path)
+        elif path.is_dir():
+            self.analysis.log.debug(f"Setting datastore from directory: {path}")
+            self.analysis.datastore = DataStore.from_dir(path)
+        else:
+            raise FileNotFoundError(f"Datastore not found: {path}")
+
+    def _make_obs_table_selection(self):
+        """Return list of obs_ids after filtering on datastore observation table."""
+        obs_settings = self.analysis.config.observations
+
+        # Reject configs with list of obs_ids and obs_file set at the same time
+        if len(obs_settings.obs_ids) and obs_settings.obs_file is not None:
+            raise ValueError(
+                "Values for both parameters obs_ids and obs_file are not accepted."
+            )
+
+        # First select input list of observations from obs_table
+        if len(obs_settings.obs_ids):
+            selected_obs_table = self.analysis.datastore.obs_table.select_obs_id(
+                obs_settings.obs_ids
+            )
+        elif obs_settings.obs_file is not None:
+            path = make_path(obs_settings.obs_file)
+            ids = list(Table.read(path, format="ascii", data_start=0).columns[0])
+            selected_obs_table = self.analysis.datastore.obs_table.select_obs_id(ids)
+        else:
+            selected_obs_table = self.analysis.datastore.obs_table
+
+        # Apply cone selection
+        if obs_settings.obs_cone.lon is not None:
+            cone = dict(
+                type="sky_circle",
+                frame=obs_settings.obs_cone.frame,
+                lon=obs_settings.obs_cone.lon,
+                lat=obs_settings.obs_cone.lat,
+                radius=obs_settings.obs_cone.radius,
+                border="0 deg",
+            )
+            selected_obs_table = selected_obs_table.select_observations(cone)
+
+        return selected_obs_table["OBS_ID"].tolist()
+    
+class DatasetsAnalysisStep(AnalysisStepBase):
```

Comment:
This is currently a very large class with lots of responsibility, maybe it could be split further...

Reply:
Yes possibly there could be a step responsible for the geometry and dataset creation and one for handling makers and filling them

Reply:
for now the analysis steps mirror the get_xx methods, I could try to further split theses classes in separate PR (one for each)